File: banda/SergioDeIscar-banda-java/SergioDeIscar-banda-java/src/main/java/org/example/models/Trompetista.java
package org.example.models;

import org.example.interfaces.ITrompetista;

public final class Trompetista extends Musician implements ITrompetista {
    private int capacidadPulmonar;
    private final float salario = super.salario * 1.2f;
    public Trompetista(String nombre, int experiencia, int capacidadPulmonar) {
        super(nombre, experiencia);
        setCapacidadPulmonar(capacidadPulmonar);
    }

    @Override
    public float getSalario() {
        return salario;
    }

    @Override
    public int getCapacidadPulmonar() {
        return capacidadPulmonar;
    }

    @Override
    public void setCapacidadPulmonar(int capacidadPulmonar) {
        this.capacidadPulmonar = capacidadPulmonar;
    }

    @Override
    public String getNombre() {
        return this.nombre;
    }
}

File: banda/SergioDeIscar-banda-kotlin/SergioDeIscar-banda-kotlin/src/main/kotlin/models/Trompetista.kt
package bandamusical.models

class Trompetista(nombre: String,
                  override val experiencia: Int,
                  val capacidadPulmonar: IntRange = 0..15): Musician(nombre) {
    override val salario: Float = super.salario * 1.2f
    override fun respirar() {
        println("El músico trompetista $nombre respira.")
        for (i in capacidadPulmonar){
            println("Coge aire")
        }
    }
}